src/transformers/pipelines/base.py

```diff
@@ -748,6 +748,8 @@ def _parse_and_tokenize(
         Parse arguments and tokenize
         """
         # Parse arguments
+        if self.tokenizer.pad_token is None:
+            padding = False
```

Comment:
This looks like a dangerous change, it would override whatever the user specified as the default `padding` value, and just not pad

Reply:
Ok for keeping it user specified if done so. I think we have some callers within the lib that do specify padding though.

Reply:
Done.

Reply:
:+1: